src/Symfony/Component/Ldap/Security/User/LdapUser.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Ldap\Security\User;
+
+use Symfony\Component\Ldap\Entry;
+use Symfony\Component\Security\Core\User\UserInterface;
+
+/**
+ * @author Robin Chalas <robin.chalas@gmail.com>
+ *
+ * @internal
```

Comment:
is this a requirement? do we anticipate possible direct use in userland?

Reply:
People have been using the `InMemoryUser` (aka `User`) in user-land as well, albeit having a more generic name, I think it's good to mention that this is (package) internal.

Reply:
yes, the security-core `User` should be internal since day 1 to me. Reasoning is that `UserInterface` should be the only core type needed  in userland. 
But, given this one exposes a `getExtraFields()` introduced not so long ago (https://github.com/symfony/symfony/pull/31532), I switched to just `@final`.

Reply:
Would it be an option to move those to the token attributes instead?

Reply:
they are about domain user info such as email address, phone number... so not sure actually. If I had to use this provider, I would probably write a user class with proper getters and build it from the LdapUser

Reply:
If you'd not have the user object outside of the inner security workings anymore, thus move it to just the domain object, how would you solve that for LDAP?